### PR TITLE
fix: fix network client ID used on the useGasFeeInputs hook

### LINKED ISSUE #28263: [Bug]: Delete Networks - When I delete Sepolia and re-add it again I get the error `No Infura network client was found with the ID "sepolia"`
### Describe the bug

When I delete Sepolia and re-add it again I get the error  `No Infura network client was found with the ID "sepolia"`.
The requests are working normally.

### Expected behavior

No error, like what's currently in prod

### Screenshots/Recordings

https://github.com/user-attachments/assets/2889042d-2cf1-4c2e-843a-4cde2902eee7

### Steps to reproduce

1. Remove Sepolia network
2. Add Sepolia network using Infura RPC
3. Switch to Sepolia
4. Click Activity tab
5. See UI console error

### Error messages or log output

_No response_

### Detection stage

During release testing

### Version

12.7.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**

When I delete Sepolia and re-add it again I get the error No Infura network client was found with the ID "sepolia".
The requests are working normally

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28391?quickstart=1)

## **Related issues**

Fixes: #28263 #28247

## **Manual testing steps**

1. process transaction using sepolia
2. delete sepolia network
3. add it and go to activity tab

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/hooks/useGasFeeInputs.js [hunk 1]
```diff
@@ -130,7 +132,7 @@ export function useGasFeeInputs(
     gasFeeEstimates,
     isGasEstimatesLoading,
     isNetworkBusy,
-  } = useGasFeeEstimates(transaction?.networkClientId);
```
Comment (salimtb) [inline ui/pages/confirmations/hooks/useGasFeeInputs.js:133]: The cause of this bug is that we've already processed a transaction with network ID X, which is included in the transaction payload. Later, we delete the network and add a new one with ID Y. However, the system still tries to use the original ID X, which no longer exists.

**Comment by bergeron** [comment]
This looks like it also resolves https://github.com/MetaMask/metamask-extension/issues/28247

File: ui/pages/confirmations/hooks/useGasFeeInputs.js (hunk 1 above)
Comment (bergeron) [reply ui/pages/confirmations/hooks/useGasFeeInputs.js:133]: In addition to deleting the entire network, the bug is also triggered when removing just the endpoint used for a previous transaction.

File: ui/pages/confirmations/hooks/useGasFeeInputs.js [hunk 2]
```diff
@@ -118,6 +119,14 @@ export function useGasFeeInputs(
     ? retryTxMeta
     : _transaction;
 
+  const network = useSelector((state) =>
+    selectNetworkConfigurationByChainId(state, transaction?.chainId),
+  );
+
+  const networkClientId =
+    network?.rpcEndpoints?.[network?.defaultRpcEndpointIndex].networkClientId ??
```
Comment (bergeron) [inline ui/pages/confirmations/hooks/useGasFeeInputs.js:127]: If we're going to use safe accessors might want to include one here too

```suggestion
    network?.rpcEndpoints?.[network?.defaultRpcEndpointIndex]?.networkClientId ??
```

Comment (salimtb) [reply ui/pages/confirmations/hooks/useGasFeeInputs.js]: done [2a5810e](https://github.com/MetaMask/metamask-extension/pull/28391/commits/2a5810e5ecd94d34030afb7b614bafdd52cd999c)

File: ui/pages/confirmations/hooks/useGasFeeInputs.js [hunk 3]
```diff
@@ -118,6 +119,14 @@ export function useGasFeeInputs(
     ? retryTxMeta
     : _transaction;
 
+  const network = useSelector((state) =>
+    selectNetworkConfigurationByChainId(state, transaction?.chainId),
+  );
+
+  const networkClientId =
+    network?.rpcEndpoints?.[network?.defaultRpcEndpointIndex]
+      ?.networkClientId ?? transaction?.networkClientId;
```
Comment (OGPoyraz) [inline ui/pages/confirmations/hooks/useGasFeeInputs.js:128]: Can we remove the `transaction?.networkClientId` completely then? There is no possibility to having a transaction without `chainId` so this makes this branching not necessary.

File: ui/pages/confirmations/hooks/useGasFeeInputs.test.js [hunk 4]
```diff
@@ -190,6 +193,17 @@ describe('useGasFeeInputs', () => {
   });
 
   describe('editGasMode', () => {
+    beforeEach(() => {
```
Comment (OGPoyraz) [inline ui/pages/confirmations/hooks/useGasFeeInputs.test.js:196]: Can we make sure adding a small unit test where it calls mocked `selectNetworkConfigurationByChainId` with given chain id?

File: ui/pages/confirmations/hooks/useGasFeeInputs.test.js [hunk 5]
```diff
@@ -50,6 +51,7 @@ jest.mock('../../../hooks/useMultichainSelector', () => ({
 const mockTransaction = {
   status: TransactionStatus.unapproved,
   type: TransactionType.simpleSend,
+  networkClientId: '1',
```
Comment (OGPoyraz) [inline ui/pages/confirmations/hooks/useGasFeeInputs.test.js:54]: This line will not have any effect in the tests.

File: ui/pages/confirmations/hooks/useGasFeeInputs.test.js (hunk 4 above)
Comment (salimtb) [reply ui/pages/confirmations/hooks/useGasFeeInputs.test.js:200]: test added [here](https://github.com/MetaMask/metamask-extension/pull/28391/files#diff-9c8c03c352032e2a5051eb34891d9f417ef712f6cf39b5aa1c30fc98986c4681R193-R196)

**Review by Prithpal-Sooriya — APPROVED** [review]

File: ui/pages/confirmations/hooks/useGasFeeInputs.js (hunk 1 above)
Comment (Prithpal-Sooriya) [reply ui/pages/confirmations/hooks/useGasFeeInputs.js:133]: Nice fix!

Would this bug have happened for custom RPCs (where the networkClientId is a uuid)?

Also just want to confirm that the `useGasFeeEstimates` hook can handle if it's params is undefined (e.g. the network was outright deleted, very unlikely in this flow?)

**Review by NidhiKJha — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**